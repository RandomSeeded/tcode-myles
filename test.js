import { encode, decode } from './index'

test('strings without newlines encoded as simple strings', () => {
  expect(encode('foo')).toEqual('+foo\n')
})

test('empty string encoded as simple string', () => {
  expect(encode('')).toEqual('+\n')
})

test('presence of newline switches to bulk string encoding', () => {
  expect(encode('foo\n')).toEqual(`$4\nfoo\n\n`)
})

test('decode null', () => {
  expect(decode('~\n')).toEqual([null, 2])
})

test('decode simple string', () => {
  expect(decode('+foo\n')).toEqual(['foo', 5])
})

test('decode bulk string', () => {
  expect(decode('$4\nbar\n\n')).toEqual(['bar\n', 8])
})

test('decode stream', () => {
  let stream = '+foo\n+bar\n~\n'
  let offset, v0, v1, v2
  ;[v0, offset] = decode(stream, offset)
  ;[v1, offset] = decode(stream, offset)
  ;[v2, offset] = decode(stream, offset)
  expect([v0, v1, v2]).toEqual(['foo', 'bar', null])
})
