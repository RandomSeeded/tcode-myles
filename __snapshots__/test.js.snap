// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decoding an array with multiple elements works 1`] = `
Array [
  Array [
    null,
    null,
  ],
  4,
]
`;

exports[`decoding an array with type null works 1`] = `
Array [
  Array [
    null,
  ],
  2,
]
`;

exports[`encoding an array of null returns expected value 1`] = `
"~
"
`;

exports[`packing an int returns them encoded 1`] = `"32"`;

exports[`packing decode works 1`] = `
Array [
  1,
  2,
]
`;

exports[`packing encode works 1`] = `
Array [
  "1",
  "2",
]
`;

exports[`unpacking an int works 1`] = `32`;

exports[`we can both encode and decode 1`] = `
"+abcd
~
+efgh
"
`;

exports[`we can both encode and decode 2`] = `
Array [
  Array [
    "abcd",
    null,
    "efgh",
  ],
  14,
]
`;
